src/pybind/mgr/rook/module.py

```diff
@@ -248,19 +248,16 @@ def get_hosts(self):
         return [orchestrator.InventoryNode(n, inventory.Devices([])) for n in self.rook_cluster.get_node_names()]
 
     @deferred_read
-    def describe_service(self, service_type=None, service_id=None, node_name=None, refresh=False):
+    def list_daemons(self, daemon_type=None, daemon_id=None, node_name=None, refresh=False):
 
-        if service_type not in ("mds", "osd", "mgr", "mon", "nfs", None):
-            raise orchestrator.OrchestratorValidationError(service_type + " unsupported")
-
-        pods = self.rook_cluster.describe_pods(service_type, service_id, node_name)
+        pods = self.rook_cluster.describe_pods(daemon_type, daemon_id, node_name)
 
         result = []
         for p in pods:
-            sd = orchestrator.ServiceDescription()
+            sd = orchestrator.DaemonDescription()
```

Comment:
I would create a new method **from_k8s_pod(pod)** in the **orchestrator.DaemonDescription** class, and move all the logic included inside this _for loop_ in order to set the DaemonDescription attributes.

The orchestrator module is generic, but this does not mean that we can have specialized methods in this module to make easy to work with the generic objects defined here. For Cephadm probably it would be nice also to have a method "from_container(container)"

In this way the implementation of **list_daemons** method could be just:

`result = [orchestrator.DaemonDescription().from_k8s_pod(pod) for pod in self.rook_cluster.describe_pods(daemon_type, daemon_id, node_name)]`